src/components/UserProfile/elements/TransitionReplace.jsx

```diff
@@ -36,20 +36,21 @@ class TransitionReplace extends React.Component {
   }
 
   onChildEntering(htmlNode) {
-    // subtract 1 from the height to improve shifts at the end of the animation
     this.setState({
-      height: htmlNode.offsetHeight - 1,
+      height: htmlNode.offsetHeight,
     });
   }
 
   onChildEntered(htmlNode) {
     if (this.props.onChildEntered) this.props.onChildEntered(htmlNode);
+    this.setState({
+      height: null,
+    });
```

Comment:
By setting the height to `null` here we are unsetting the height (which should always be null when there is no transition happening) after a child has finished entering. This is necessary now that we can't rely on a childExited event, given that the content of this component can be empty.

Reply:
Is the above worth adding as a comment in the code?

Reply:
The statement applies to the component as a whole, not just that line. Do you think it's worth describing how the component works at the top of the file?